src/assets/wise5/classroomMonitor/dataExport/data-export/data-export.component.ts

```diff
@@ -1190,18 +1188,127 @@ export class DataExportComponent implements OnInit {
         }
+      });
+    });
+  }
+
+  private insertMatchChoiceReuseValues(
+    row: string[],
+    columnNameToNumber: any,
+    component: any,
+    matchComponentState: any
+  ): void {
+    matchComponentState.studentData.buckets
+      .filter((bucket: Bucket) => bucket.id !== '0')
+      .forEach((bucket: Bucket) => {
+        bucket.items.forEach((item: Choice) => {
+          this.setBucketValueWithAppending(row, columnNameToNumber, component, bucket, item);
+          if (
+            this.includeCorrectnessColumns &&
+            this.matchService.componentHasCorrectAnswer(component)
+          ) {
+            this.setCorrectnessValueForChoiceReuse(row, columnNameToNumber, item);
+          }
+        });
+      });
+  }
+
+  private setBucketValueWithAppending(
+    row: string[],
+    columnNameToNumber: any,
+    component: any,
+    bucket: any,
+    item: any
+  ): void {
+    if (row[columnNameToNumber[item.id]] == null || row[columnNameToNumber[item.id]] === '') {
+      row[columnNameToNumber[item.id]] = this.getBucketValueById(component, bucket.id);
+    } else {
+      row[columnNameToNumber[item.id]] = `${
+        row[columnNameToNumber[item.id]]
+      }, ${this.getBucketValueById(component, bucket.id)}`;
+    }
+  }
+
+  private setCorrectnessValueForChoiceReuse(row: any[], columnNameToNumber: any, item: any): void {
+    const columnName = item.id + '-boolean';
+    if (item.isCorrect == null) {
+      // The item does not have an isCorrect field so we will not show anything in the cell.
+    } else if (item.isCorrect) {
+      this.mergeCorrectnessValue(row, columnNameToNumber, columnName, 1);
+    } else {
+      if (item.isIncorrectPosition) {
+        this.mergeCorrectnessValue(row, columnNameToNumber, columnName, 2);
+      } else {
+        this.mergeCorrectnessValue(row, columnNameToNumber, columnName, 0);
       }
     }
-    return row;
+  }
+
+  /**
+   * Matrix to determine the merged correctness value.
+   * Legend
+   * e = empty
+   * 0 = incorrect
+   * 1 = correct
+   * 2 = correct bucket but incorrect position
+   *        previous
+   *        e 0 1 2
+   *       --------
+   * n  0 | 0 0 0 0
+   * e  1 | 1 0 1 2
+   * w  2 | 2 0 2 2
+   * @param row: any[]
+   * @param columnNameToNumber: any
+   * @param columnName: string
+   * @param newValue: number
+   */
+  private mergeCorrectnessValue(
+    row: any,
+    columnNameToNumber: any,
+    columnName: string,
+    newValue: any
+  ): void {
+    const previousValue = row[columnNameToNumber[columnName]];
+    if (previousValue === '') {
+      row[columnNameToNumber[columnName]] = newValue;
+    } else if (this.bothValuesAreX(previousValue, newValue, 1)) {
+      row[columnNameToNumber[columnName]] = 1;
+    } else if (this.eitherValuesAreX(previousValue, newValue, 0)) {
+      row[columnNameToNumber[columnName]] = 0;
+    } else if (this.eitherValuesAreX(previousValue, newValue, 2)) {
```

Comment:
Do this check inline and remove extra function call?
```suggestion
    } else if ([previousValue, newValue].includes(2))) {
```